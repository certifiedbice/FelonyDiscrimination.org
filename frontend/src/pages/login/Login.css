#forgotten-login-info-link:hover {
    color: #e6e7e8;
}