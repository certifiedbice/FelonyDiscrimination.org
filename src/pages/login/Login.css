#login-form-container {
    flex: 1;
    text-align: center;
}

#login-form {}

#login-form legend {
    width: 100%;
    margin-bottom: 5px;
    font-size: 18px;
    font-weight: 700;
}

#login-form legend a {
    color: #4378bc;
}

#login-form legend a:hover {
    color: #e6e7e8;
}

.login-form-element-container {
    padding: 5px;   
}

.login-form-element-container label {
    display: block;
    margin: 0 0 5px 20px;
    text-align: left;
    font-size: 12px;
}

.login-form-element {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

.login-form-element-container input {
    border-radius: 50px;
    padding: 5px;
}

#login-form-submit {
    width: 90%;
    background-color: #4378bc;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
    cursor: pointer;
}

#login-form-submit:hover {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

#forgotten-login-info-link:hover {
    color: #e6e7e8;
}

/* Desktop CSS */
@media(min-width:991px){
    
    .login-form-element-container label {
        margin: 0 0 10px 60px;
    }

    #login-form-submit {
        width: 25%;
    }

    #login-form-submit:hover {
        width: 25%;
    }
    
}