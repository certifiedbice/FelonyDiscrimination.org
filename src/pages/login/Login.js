import React from 'react';
import {Link} from 'react-router-dom';
import './Login.css';

export default function Login(){
	return(
		<section id="login-form-container">
			<form id="login-form" name="login-form" aria-label="Login form">
				<fieldset>
					<legend><h2>Login or <Link to='/signup'>Sign Up</Link></h2></legend>
					<div className="login-form-element-container">
						<label>Email: </label>
						<input className="login-form-element" name="login-form-email" id="login-form-email" type="text" required aria-labelledby="Login form email" placeholder="Email"/>
					</div>
					<div className="login-form-element-container">
						<label>Password: </label>
						<input className="login-form-element" name="login-form-password" id="login-form-password" type="password" required aria-labelledby="Login form password" placeholder="Password"/>
					</div>
					<div className="login-form-error-container">
					</div>
					<div className="login-form-element-container">
						<input id="login-form-submit" type="submit" required aria-label="Login form submit" value="Submit"/>
					</div>
				</fieldset>
			</form>
			<h3><Link id='forgotten-login-info-link' to='/forgottenlogininfo'>Forgotten your Login info?</Link></h3>			
		</section>
	);
}