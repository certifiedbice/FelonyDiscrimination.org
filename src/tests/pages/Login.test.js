import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Login from '../../pages/login/Login';

it('renders without crashing',()=>{
    const section=document.createElement('section');
    ReactDOM.render(<BrowserRouter><Login/></BrowserRouter>,section);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<BrowserRouter><Login/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});