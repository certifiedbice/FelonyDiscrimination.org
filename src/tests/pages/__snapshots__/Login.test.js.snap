// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="login-form-container"
>
  <form
    aria-label="Login form"
    id="login-form"
    name="login-form"
  >
    <fieldset>
      <legend>
        <h2>
          Login or 
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign Up
          </a>
        </h2>
      </legend>
      <div
        className="login-form-element-container"
      >
        <label>
          Email: 
        </label>
        <input
          aria-labelledby="Login form email"
          className="login-form-element"
          id="login-form-email"
          name="login-form-email"
          placeholder="Email"
          required={true}
          type="text"
        />
      </div>
      <div
        className="login-form-element-container"
      >
        <label>
          Password: 
        </label>
        <input
          aria-labelledby="Login form password"
          className="login-form-element"
          id="login-form-password"
          name="login-form-password"
          placeholder="Password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="login-form-error-container"
      />
      <div
        className="login-form-element-container"
      >
        <input
          aria-label="Login form submit"
          id="login-form-submit"
          required={true}
          type="submit"
          value="Submit"
        />
      </div>
    </fieldset>
  </form>
  <h3>
    <a
      href="/forgottenlogininfo"
      id="forgotten-login-info-link"
      onClick={[Function]}
    >
      Forgotten your Login info?
    </a>
  </h3>
</section>
`;
